build.gradle

```diff
@@ -468,6 +468,79 @@ project(':iceberg-mr') {
   }
 }
 
+if (jdkVersion == '8') {
+  project(':iceberg-mr-hive3') {
+
+    // run the tests in iceberg-mr with Hive3 dependencies
+    sourceSets {
+      test {
+        java.srcDirs = ['../mr/src/test/java', 'src/test/java']
+        resources.srcDirs = ['../mr/src/test/resources', 'src/test/resources']
+      }
+    }
+
+    // exclude these Hive2-specific tests from iceberg-mr
+    test {
+      exclude '**/TestIcebergDateObjectInspector.class'
+      exclude '**/TestIcebergTimestampObjectInspector.class'
+    }
+
+    configurations.all {
+      resolutionStrategy.eachDependency { dep ->
+        if (dep.requested.group == 'org.apache.hive' && dep.requested.name != 'hive-storage-api') {
+          dep.useVersion '3.1.2'
+        } else if (dep.requested.group == 'org.apache.hadoop') {
+          dep.useVersion '3.1.0'
+        }
+      }
+    }
```

Comment:
Is this block needed? I think we should be able to explicitly set dependency versions in the `dependencies` block, like this:

```
compileOnly("org.apache.hive:hive-exec:3.1.2:core") {
  ...
}

compileOnly("org.apache.hadoop:hadoop-client:3.1.0") {
  ...
}
```

Reply:
yes, you're right, don't need. I've set the versions explicitly instead